HLTrigger/JetMET/plugins/HLTHFJetCleaner.cc

```diff
@@ -0,0 +1,268 @@
+//
+
+template <typename T>
+std::vector<T> HLTHFJetCleaner::extractValueMapParameters(edm::Handle<edm::ValueMap<T>> valueMapHandle,
+                                                          size_t numJets) {
+  // Extract the values as a vector from a given ValueMap object.
+  std::vector<T> values;
+  values.reserve(numJets);
+  for (auto it = valueMapHandle->begin(); it != valueMapHandle->end(); it++) {
+    auto productId = it.id();
+    for (size_t i = 0; i < numJets; i++) {
+      T value = valueMapHandle->get(productId, i);
+      values.push_back(value);
+    }
+  }
+  return values;
+}
+
+// ------------ method called to produce the data  ------------
+void HLTHFJetCleaner::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  edm::Handle<edm::View<reco::PFJet>> jets;
+  edm::Handle<edm::View<reco::CaloMET>> mets;
+
+  edm::Handle<edm::ValueMap<float>> sigmaEtaEtas;
+  edm::Handle<edm::ValueMap<float>> sigmaPhiPhis;
+  edm::Handle<edm::ValueMap<int>> centralEtaStripSizes;
+
+  iEvent.getByToken(jetToken_, jets);
+
+  iEvent.getByToken(sigmaEtaEtaToken_, sigmaEtaEtas);
+  iEvent.getByToken(sigmaPhiPhiToken_, sigmaPhiPhis);
+  iEvent.getByToken(centralEtaStripSizeToken_, centralEtaStripSizes);
+
+  // The set of cleaned jets
+  std::unique_ptr<JetCollection> cleaned_jets(new JetCollection);
+
+  std::vector<float> v_sigmaEtaEta = extractValueMapParameters<float>(sigmaEtaEtas, jets->size());
+  std::vector<float> v_sigmaPhiPhi = extractValueMapParameters<float>(sigmaPhiPhis, jets->size());
+  std::vector<int> v_centralEtaStripSize = extractValueMapParameters<int>(centralEtaStripSizes, jets->size());
+
+  // Loop over the jets and do cleaning
+  for (size_t ijet = 0; ijet < jets->size(); ijet++) {
+    reco::PFJet jet = (*jets)[ijet];
+    bool withinEtaRange = (abs(jet.eta()) < jetEtaMax_) & (abs(jet.eta()) > jetEtaMin_);
+    // Cuts are applied to jets that are back to back with MET
+    iEvent.getByToken(metToken_, mets);
+    const float met_phi = mets->front().phi();
+    bool backToBackWithMet = deltaPhi(jet.phi(), met_phi) > dphiJetMetThresh_;
+    if (!backToBackWithMet) {
+      cleaned_jets->push_back(jet);
+      continue;
+    }
+    // If the jet is outside the range that we're interested in,
+    // append it to the cleaned jets collection and continue
+    if ((jet.pt() < jetPtThresh_) | (!withinEtaRange)) {
+      cleaned_jets->push_back(jet);
+      continue;
+    }
+    // Cuts related to eta and phi width of HF showers:
+    // Sigma eta-phi based cut + central strip size cut
+    // We'll apply the sigma eta+phi dependent cut only if requested via the config file
+    if (applySigmaEtaPhiCut_) {
+      float sieie = v_sigmaEtaEta[ijet];
+      float sipip = v_sigmaPhiPhi[ijet];
+
+      // Sigma eta eta and sigma phi phi
+      bool passSigmaEtaPhiCut = sieie - sipip < sigmaEtaPhiDiffThresh_;
+
+      if (applyCornerCut_) {
+        bool inCorner = (v_sigmaEtaEta[ijet] < 0.02) & (v_sigmaPhiPhi[ijet] < 0.02);
```

Comment:
`0.02` is hard-coded. Is it well-motived to hard-code it (e.g. immutable value related to the detector)? If not, please make it configurable via the ParameterSet.

Reply:
The `0.02` value was motivated by the analysis we did on these HF shape variables, you can see slide 10 [here](https://indico.cern.ch/event/1125473/contributions/4724108/attachments/2391723/4088717/2022-02-15_hlt_exo.pdf). But I agree that in principle it would be better to make it configurable via the ParameterSet, I'll update that.